src/charm.py

```diff
@@ -218,6 +238,24 @@ def server_cert_path(self) -> Optional[str]:
     # === UTILITY METHODS === #
     ###########################
 
+    def _update_mimir_cluster(self):  # common exit hook
+        # TODO: does this prevent disabling log forwarding on relation-broken ???
+        # if not self.coordinator.is_coherent():
```

Comment:
this is also applicable imho. why is it commented out?

Reply:
note the previous implementation had this guard and:
```
logger.warning("Incoherent deployment: Some required Mimir roles are missing.")
```

I vote we keep it

Reply:
Right, so this is not a problem anymore because the worker will (by itself) clear the log forwarding endpoints on relation-broken, instead of relying on the coordinator to empty the databag.